discovery-provider/src/model_validator.py

```diff
@@ -0,0 +1,174 @@
+import os.path
+import json
+import copy
+import logging  # pylint: disable=C0302
+from jsonschema import Draft7Validator, ValidationError
+logger = logging.getLogger(__name__)
+
+# https://app.quicktype.io/ -- JSON schema generator
+class ModelValidator:
+    '''
+        {key: value} :
+        {
+            model (first letter uppercase): {
+                model_schema: schema,
+                field_schema: {
+                    title: schema,
+                    mood: schema,
+                    ...
+            },
+                fields: [title, mood, ...]
+            }
+        }
+    '''
+    models_to_schema_and_fields_dict = {}
+    BASE_PATH = './src/schemas/'
+    FILE_NAME = ''
+
+    @classmethod
+    def validate(cls, to_validate, model, field=""):
+        cls.FILE_NAME = model.lower() + '_schema.json'
+        try:
+            schema = cls.get_schema(field, model)
+            validator = Draft7Validator(schema)
+
+            found_invalid_field = False
+            for error in sorted(validator.iter_errors(to_validate), key=str):
+                found_invalid_field = True
+                logger.warning(f"Error with {model} instance {to_validate}: {error.message}")
+
+            # if any error occurs, raise exception
+            if found_invalid_field:
+                raise ValidationError(f"Instance {to_validate} is not proper")
+
+        except ValidationError as ve:
+            raise ve
+        except BaseException as e:
+            # one of many errors specified in helper methods
+            raise e
+
+    @classmethod
+    def get_schema(cls, field, model):
```

Comment:
i don't think your suggested name describes the method function. this function returns the schema for the entire model or just a field and is used in `validate()` for validation

Reply:
sorry i think i got that backwards. how's like get_schema_for_field?

Reply:
that works! what's the reason behind this suggestion? i thought get_schema() was fitting because it accurately describes the intention

Reply:
it's just confusing. there's no comment saying that that field is optional, there's no default value set for field and it can take an empty string value for the whole schema? it was just confusing to understand the intention because the name implies return the schema, but there's a field option too

Reply:
that makes sense. i updated the method name and added a comment about the field option